src/amount.cpp

```diff
@@ -9,16 +9,18 @@
 
 const std::string CURRENCY_UNIT = "BTC";
 
-CFeeRate::CFeeRate(const CAmount& nFeePaid, size_t nSize)
+CFeeRate::CFeeRate(const CAmount& nFeePaid, size_t nSize_)
 {
+    int64_t nSize = nSize_;
     if (nSize > 0)
-        nSatoshisPerK = nFeePaid*1000/nSize;
+        nSatoshisPerK = nFeePaid * 1000 / nSize;
```

Comment:
Instead of the temporary variable, why not just:

    nSatoshisPerK = nFeePaid * 1000 / (int64_t)nSize;

Also at least theoretically you should handle the case where nSize is outside the range of `int64_t`, though I doubt someone will ever care about transactions that large.